app/app.go

```diff
@@ -120,11 +112,23 @@ func (a *app) GetDeviceIntegrations(
 	ctx context.Context,
 	deviceID string,
 ) ([]model.Integration, error) {
-	// TODO: stub only, needs to be implemented
+	device, err := a.store.GetDevice(ctx, deviceID)
+	if err != nil {
+		if err == store.ErrObjectNotFound {
+			return nil, ErrDeviceNotFound
+		}
+		return nil, errors.Wrap(err, "app: failed to get device integrations")
+	}
+	if len(device.IntegrationIDs) > 0 {
+		integrations, err := a.store.GetIntegrations(ctx,
+			model.IntegrationFilter{IDs: device.IntegrationIDs},
+		)
+		return integrations, errors.Wrap(err, "app: failed to get device integrations")
+	}
 	return []model.Integration{}, nil
 }
 
-func (a *app) SetDeviceStatus(ctx context.Context, deviceID string, status Status) error {
+func (a *app) SetDeviceStatus(ctx context.Context, deviceID string, status model.Status) error {
```

Comment:
Nitpick: please differentiate Mender device status from Azure device status: 
```
SetDeviceStatus(ctx context.Context, deviceID string, menderStatus model.Status)
```

Reply:
@MaciejTe feel free to address this in your PR